spec.html

```diff
@@ -21026,16 +21232,287 @@ <h1>Runtime Semantics: Evaluation</h1>
+          1. Perform ! CreateDataPropertyOrThrow(_accessObj_, *"get"*, _setter_).
+        1. Return _accessObj_.
+      </emu-alg>
+      <p>A PrivateValueGetter function is an anonymous built-in function with a [[PrivateStateKey]] internal slot. When a PrivateValueGetter function that expects no arguments is called it performs the following steps:</p>
+      <emu-alg>
+        1. Let _o_ be the *this* value.
+        1. Let _f_ be the active function object.
+        1. Return ? PrivateGet(_f_.[[PrivateStateKey]], _o_).
+      </emu-alg>
+      <p>A PrivateValueSetter function is an anonymous built-in function with a [[PrivateStateKey]] internal slot. When a PrivateValueSetter function is called with argument _value_ it performs the following steps:</p>
+      <emu-alg>
+        1. Let _o_ be the *this* value.
+        1. Let _f_ be the active function object.
+        1. Perform ? PrivateSet(_f_.[[PrivateStateKey]], _o_, _value_).
+        1. Return *undefined*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createsetmetadatafunction" aoid="CreateSetMetadataFunction">
+      <h1>CreateSetMetadataFunction ( _kind_, _metadataList_, _propertyKey_ )</h1>
+      <p>
+        The abstract operation CreateSetMetadataFunction takes arguments _kind_ (one of ~constructor~, ~public~, ~private~), _metadataList_ (a List of Metadata Records), and _propertyKey_ (a String, Symbol, Private Name, or ~empty~). It returns a function that can be used to define metadata on the passed in _metadataList_, and which is associated with the given _kind_ and, if provided, _propertyKey_.
+      </p>
+      <emu-alg>
+        1. Let _steps_ be the steps of a SetDecoratorMetadata function as specified below.
+        1. Let _length_ be the number of non-optional parameters of a SetDecoratorMetadata function as specified below.
+        1. Let _defineMetadata_ be ! CreateBuiltinFunction(_steps_, _length_, *""*, &laquo; [[Kind]], [[MetadataList]], [[PropertyKey]] &raquo;).
+        1. Set _defineMetadata_.[[Kind]] to _kind_.
+        1. Set _defineMetadata_.[[MetadataList]] to _metadataList_.
+        1. Set _defineMetadata_.[[PropertyKey]] to _propertyKey_.
+        1. Return _defineMetadata_.
+      </emu-alg>
+      <p>A SetDecoratorMetadata function is an anonymous built-in function with [[Kind]], [[MetadataList]], and [[PropertyKey]] internal slots. When a SetDecoratorMetadata function is called with arguments _metadataKey_ and _value_, the following steps are taken:</p>
+      <emu-alg>
+        1. If _metadataKey_ is not a Symbol, throw a *TypeError* exception.
+        1. Let _f_ be the active function object.
+        1. Let _kind_ be _f_.[[Kind]].
+        1. Let _metadataList_ be _f_.[[MetadataList]].
+        1. Let _propertyKey_ be _f_.[[PropertyKey]].
+        1. Let _metadataForKey_ be the element of _metadataList_ whose [[Key]] is the same as _metadataKey_.
+        1. If _metadataForKey_ does not exist, then
```

Comment:
A similar idiom exists in `ParseModule` and elsewhere. I originally asked about adding a Map construct to complement the List construct, since that is what is happening here, but after discussion with the editors it seemed like they preferred continuing to build on the List construct in this way.

Reply:
What do you think is a similar idiom in `ParseModule`?

Reply:
```
1. Let _ie_ be the element of _importEntries_ whose [[LocalName]] is the same as _ee_.[[LocalName]].
1. If _ie_.[[ImportName]] is *"\*"*, then
  1. NOTE: This is a re-export of an imported module namespace object.
  1. Append _ee_ to _localExportEntries_.
1. Else,
  1. NOTE: This is a re-export of a single name.
  1. Append the ExportEntry Record { [[ModuleRequest]]: _ie_.[[ModuleRequest]], [[ImportName]]: _ie_.[[ImportName]], [[LocalName]]: *null*, [[ExportName]]: _ee_.[[ExportName]] } to _indirectExportEntries_.
```

Reply:
I'm fine with `the element of <List> whose <field_name> is the same as <expr>`. But in `ParseModule`, such an element *exists*, so `_ie_` is well-defined. In your algorithm, there might not be such an element, so `_metadataForKey_` might be an alias for a lack of a result, which you check via `If _metadataForKey_ does not exist`. I'm saying that nowhere in the spec do we have `If <alias> does not exist` or `If <alias> exists`.